packages/eas-cli/src/build/utils/repository.ts

```diff
@@ -129,19 +129,26 @@ async function makeProjectTarballAsync(): Promise<{ path: string; size: number }
     await tar.create({ cwd: shallowClonePath, file: tarPath, prefix: 'project', gzip: true }, [
       '.',
     ]);
-  } finally {
-    // Stop the timer
+  } catch (err) {
     clearTimeout(timer);
-    await fs.remove(shallowClonePath);
-
-    const duration = endTimer(compressTimerLabel);
     if (spinner.isSpinning) {
-      const prettyTime = formatMilliseconds(duration);
-      spinner.succeed(`Compressed project files ${chalk.dim(prettyTime)}`);
+      spinner.fail();
     }
+    throw err;
+  } finally {
+    await fs.remove(shallowClonePath);
   }
+  clearTimeout(timer);
 
   const { size } = await fs.stat(tarPath);
+  const duration = endTimer(compressTimerLabel);
+  if (spinner.isSpinning) {
+    const prettyTime = formatMilliseconds(duration);
+    spinner.succeed(
+      `Compressed project files into ${formatBytes(size)} tar archive. ${chalk.dim(prettyTime)}`
```

Comment:
```suggestion
      `Compressed project files ${chalk.dim(prettyTime)} (${formatBytes(size)})`
```
(According to https://linear.app/expo/issue/ENG-399/log-the-size-of-the-archived-project-when-uploading)